src/plugin.js

```diff
@@ -155,8 +155,10 @@ Chart.plugins.register({
 			if (options.delay > 0) {
 				model.delayed = true;
 				defer(function() {
-					model.delayed = false;
-					chart.update();
+					if (chart.ctx !== null && chart.canvas !== null) {
```

Comment:
I think it's enough to test `chart.ctx` because if the context is valid, the canvas should also be valid. Though, I'm not able to reproduce this issue so can you check if it's indeed the case?

Also, can you add a comment with the link to the related issue:

```js
// https://github.com/chartjs/chartjs-plugin-deferred/issues/13
if (chart.ctx) {
  model.delayed = false;
  chart.update();
}
```

Reply:
Most probably that should be enough however I see no big difference in just a single additional condition.

I'll check whether the simplified version works too.

Reply:
Checking just for `chart.ctx` is enough. In the example below none of the asserts failed.

```html
<html>
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, user-scalable=no, initial-scale=1.0, maximum-scale=1.0, minimum-scale=1.0" />
    <meta http-equiv="X-UA-Compatible" content="ie=edge" />
    <title>Test</title>
</head>
<body>
<canvas id="chart1"></canvas>
<script src="https://cdn.jsdelivr.net/npm/chart.js@2"></script>
<script src="https://cdn.jsdelivr.net/npm/chartjs-plugin-deferred@1"></script>
<script type="module">
  const deferredPlugin = Chart.plugins.getAll().find((plugin) => plugin.id === 'deferred');
  const originalBeforeDatasetsUpdate = deferredPlugin.beforeDatasetsUpdate;
  const chart1 = document.getElementById('chart1');
  const delay = 500;

  deferredPlugin.beforeDatasetsUpdate = (chart, options) => {
    console.assert(chart.ctx !== null, 'The `ctx` is `null`');
    console.assert(chart.canvas !== null, 'The `canvas` is `null`');

    setTimeout(() => {
      console.assert(chart.ctx === null, 'The `ctx` is not `null`');
      console.assert(chart.canvas === null, 'The `canvas` is not `null`');
    }, delay);

    originalBeforeDatasetsUpdate(chart, options);
  };

  const chart = new Chart(chart1.getContext('2d'), {
    type: 'line',
    data: {
      labels: ['January', 'February', 'March', 'April', 'May', 'June', 'July'],
      datasets: [{
        label: 'My First dataset',
        backgroundColor: 'rgb(255, 99, 132)',
        borderColor: 'rgb(255, 99, 132)',
        data: [0, 10, 5, 2, 20, 30, 45],
      }],
    },
    options: {
      plugins: {
        deferred: {
          delay,
        },
      },
    },
  });

  setTimeout(() => chart.destroy(), delay / 2);
</script>
</body>
</html>
```

Reply:
@simonbrunel the PR is updated accordingly.